source/arduino.h

```diff
@@ -552,12 +552,109 @@ void tone(int pin, unsigned int frequency, unsigned long duration);
 void noTone(int pin);
 
 //
+// Interrupt functions.
+//
+
+/// Attach a callback routine to a GPIO interrupt.
+/**
+\param[in] pin The number of the board pin for which interrupts are wanted.
+\param[in] fund The function to be called when an interrupt occurs for the specified pin.
+\param[in] mode The type of pin state changes that should cause interrupts.
+*/
+inline void attachInterrupt(uint8_t pin, std::function<void(void)> func, int mode)
+{
+    HRESULT hr;
+
+    hr = g_pins.verifyPinFunction(pin, FUNC_DIO, BoardPinsClass::NO_LOCK_CHANGE);
+
+    if (FAILED(hr))
+    {
+        ThrowError(hr, "Error occurred verifying pin: %d function: DIGITAL_IO, Error: %08x", pin, hr);

```

Comment:
This is not new behavior.  We highlight conditions that indicate the code is not doing what the user expects/wants by throwing exceptions.  Given that the code may be controlling physical things, I think this is a good idea.